plugins/repository-gcs/src/main/java/org/opensearch/repositories/gcs/GoogleCloudStorageService.java

```diff
@@ -70,6 +71,16 @@ public class GoogleCloudStorageService {
      */
     private volatile Map<String, Storage> clientCache = emptyMap();
 
+    final private GoogleApplicationDefaultCredentials googleApplicationDefaultCredentials;
+
+
+    public GoogleCloudStorageService() {
+        this.googleApplicationDefaultCredentials = new GoogleApplicationDefaultCredentials();
+    }
+    public GoogleCloudStorageService(GoogleApplicationDefaultCredentials googleApplicationDefaultCredentials) {
```

Comment:
Might be a stupid question as i am not familar with how it works, but if it is default cred (which I assume won't change for that storage), what is the point of this method that takes in GADC object?

Reply:
this constructor has been created to cater the test cases, the GoogleApplicationDefaultCredentials is created so that we can mock the calling of static methods